### PR TITLE
GPU profiling improvements

### PR DESCRIPTION
**Pull request by ChristopherManthei**
- Added a new sub-section for Unity GPU profiling that lists the various options you have and goes into a little bit more detail.
  - Fix link to old Unity profiler guide.
  - Added note that you have to disable **Graphics Jobs** in the Unity to make the GPU profiling work.
  - Added new section for HoloLens frame rate overlay.
  - Added new section for PIX.
- Updated PIX guide
  - New images to reflect latest version.
  - Added prerequisites section.
  - Documented how to profile DX11.
  - Added links to PIX capture guide.

### CODE REVIEW

**Comment by PRMerger19** [comment]
@ChristopherManthei : Thanks for your contribution! The author(s) have been notified to review your proposed change. @hferrone

File: mixed-reality-docs/mr-dev-docs/develop/unity/performance-recommendations-for-unity.md [hunk 1]
```diff
@@ -24,23 +24,45 @@ The most important first step when optimizing performance of mixed reality apps
...
+Unity's [Frame Debugger](https://docs.unity3d.com/Manual/FrameDebugger.html) is also a powerful and insightful tool to use. It will give you a good overview of what the GPU is doing each frame. Things to look out for are additional rendering targets and blit commands to copy between them as these are very expensive on HoloLens. Ideally, no off-screen render targets should be used on HoloLens. These are usually added when enabling expensive rendering features like full screen effects, MSAA, HDR etc which should be avoided.
+
+#### HoloLens frame rate overlay
+
+The Device Portal [System Performance](../advanced-concepts/using-the-windows-device-portal.md#system-performance) page has a good summary of CPU and GPU performance of the device.
+You can enable **Display frame rate counter in headset** and **Display frame rate graph in headset**. These options will enable an FPS counter and graph respectively that will give you immediate feedback in any running application on your device.
+
+#### PIX
+
+[PIX](https://devblogs.microsoft.com/pix/download/) can be used to profile Unity applications as well. There are also detailed
```
Comment (vtieto) [inline mixed-reality-docs/mr-dev-docs/develop/unity/performance-recommendations-for-unity.md:56]:
```suggestion
[PIX](https://devblogs.microsoft.com/pix/download/) can be used to profile Unity applications as well. There are also detailed instructions on how to use and install [PIX for HoloLens 2](../advanced-concepts/installing-pix-hololens.md).
```

File: mixed-reality-docs/mr-dev-docs/develop/unity/performance-recommendations-for-unity.md [hunk 2]
```diff
@@ -24,23 +24,45 @@ The most important first step when optimizing performance of mixed reality apps
...
+instructions how to use and install [PIX for HoloLens 2](../advanced-concepts/installing-pix-hololens.md).
```
Comment (vtieto) [inline mixed-reality-docs/mr-dev-docs/develop/unity/performance-recommendations-for-unity.md:57]:
```suggestion
```

File: mixed-reality-docs/mr-dev-docs/develop/unity/performance-recommendations-for-unity.md [hunk 3]
```diff
@@ -24,23 +24,45 @@ The most important first step when optimizing performance of mixed reality apps
...
+In a development build, the same scopes that you see in Unity's [Frame Debugger](https://docs.unity3d.com/Manual/FrameDebugger.html) will be shown in PIX as well and can be inspected and profiled in more detail.
+
+>[!NOTE]
+>Unity provides the ability to easily modify the render target resolution of your application at runtime through the *[XRSettings.renderViewportScale](https://docs.unity3d.com/ScriptReference/XR.XRSettings-renderViewportScale.html)* property. The final image presented on device has a fixed resolution. The platform will sample the lower resolution output to build a higher resolution image for rendering on displays.
```
Comment (vtieto) [inline mixed-reality-docs/mr-dev-docs/develop/unity/performance-recommendations-for-unity.md:61]:
```suggestion
>Unity provides the ability to easily modify the render target resolution of your application at runtime through the *[XRSettings.renderViewportScale](https://docs.unity3d.com/ScriptReference/XR.XRSettings-renderViewportScale.html)* property. The final image presented on-device has a fixed resolution. The platform will sample the lower resolution output to build a higher resolution image for rendering on displays.
```

**Review by vtieto — COMMENTED** [review]
A few minor tweaks--otherwise, looks great!

**Comment by PRMerger12** [comment]
@ChristopherManthei : Thanks for your contribution! The author(s) have been notified to review your proposed change. @hferrone

**Comment by PRMerger19** [comment]
@ChristopherManthei : Thanks for your contribution! The author(s) have been notified to review your proposed change. @hferrone

**Comment by PRMerger12** [comment]
@ChristopherManthei : Thanks for your contribution! The author(s) have been notified to review your proposed change. @hferrone

**Comment by PRMerger16** [comment]
@ChristopherManthei : Thanks for your contribution! The author(s) have been notified to review your proposed change. @hferrone

**Comment by PRMerger12** [comment]
@ChristopherManthei : Thanks for your contribution! The author(s) have been notified to review your proposed change. @hferrone

**Comment by PRMerger18** [comment]
@ChristopherManthei : Thanks for your contribution! The author(s) have been notified to review your proposed change. @hferrone

File: mixed-reality-docs/mr-dev-docs/develop/unity/performance-recommendations-for-unity.md [hunk 4]
```diff
@@ -24,23 +24,45 @@ The most important first step when optimizing performance of mixed reality apps
...
+
+![Unity CPU vs GPU](images/unity-profiler-cpu-gpu.png)
 
 >[!NOTE]
-> With the Unity Profiler connected and after adding the GPU profiler (see *Add Profiler* in top right corner), one can see how much time is being spent on the CPU & GPU respectively in the middle of the profiler. This allows the developer to get a quick approximation if their application is CPU or GPU bounded.
+> To use GPU profiling, you need to disable **Graphics Jobs** in the Unity **Player Settings**. See Unity's **[GPU Usage Profiler module](https://docs.unity3d.com/Manual/Profiler.html)** for more details.
+
+#### Unity frame debugger
+
+Unity's [Frame Debugger](https://docs.unity3d.com/Manual/FrameDebugger.html) is also a powerful and insightful tool to use. It will give you a good overview of what the GPU is doing each frame. Things to look out for are additional rendering targets and blit commands to copy between them as these are very expensive on HoloLens. Ideally, no off-screen render targets should be used on HoloLens. These are usually added when enabling expensive rendering features like full screen effects, MSAA, HDR etc which should be avoided.
```
[the remark this answers is not in the document]
Comment (ChristopherManthei) [reply mixed-reality-docs/mr-dev-docs/develop/unity/performance-recommendations-for-unity.md:47]: @vtieto thank you for all the great suggestions! This one is a bit tricky though. MSAA and HDR are not full-screen effects. An example of a full-screen effect would be Bloom, SSAO, color grading, these kinds of things. Full screen effects, MSAA, HDR are all examples of expensive rendering features which will require an additional blit in most cases.
I'll change it for now to: "These are usually added when enabling expensive rendering features (for example MSAA, HDR or full-screen effects like bloom) which should be avoided."
Let me know what you think, thanks!

**Comment by PRMerger16** [comment]
@ChristopherManthei : Thanks for your contribution! The author(s) have been notified to review your proposed change. @hferrone

**Pull request merged**